drivers/firmware/nrf_ironside/cpuconf.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+
+#include <nrfx.h>
+
+#include <zephyr/drivers/firmware/nrf_ironside/call.h>
+#include <zephyr/drivers/firmware/nrf_ironside/cpuconf.h>
+
+int ironside_cpuconf(NRF_PROCESSORID_Type cpu, void *vector_table, bool cpu_wait, uint8_t *msg,
+		     size_t msg_size)
+{
+	int err;
+	struct ironside_call_buf *const buf = ironside_call_alloc();
+
+	buf->id = IRONSIDE_CALL_ID_CPUCONF_V0;
+
+	buf->args[IRONSIDE_CPUCONF_SERVICE_CPU_IDX] = cpu;
+	buf->args[IRONSIDE_CPUCONF_SERVICE_VECTOR_TABLE_IDX] = (uint32_t)vector_table;
+	buf->args[IRONSIDE_CPUCONF_SERVICE_CPU_WAIT_IDX] = cpu_wait;
```

Comment:
this is using a full uint32_t for a single boolean, maybe this could be a `IRONSIDE_CPUCONF_SERVICE_FLAGS_IDX` and then setting just a single bit for cpu_wait? I can't predict if we will have anything else but that would make it backwards compatible if we ever want to extend the command and is a good basic pattern (collected flags parameter instead of single parameter for each flag)

Reply:
What is the cost if we never need another flag? A single & operation to apply the mask? The way it is now any bit can be set to be 1 in the protocol for this to be evaluated to true, so if we ever want to introduce a new flag we can't do it in this `cpu_wait_idx` field without having to bump to a new version of the cpuconf operation.

Reply:
The cost is an unnecessary abstraction/indirection/container for the reader to understand.

I can update the server to just check the first bit to allow easy introduction of new flags?